### PR TITLE
Fix issue relation loosing extra options on ctb update 

### LINKED ISSUE #3409: Many-to-Many relations creating table name too long
**Describe the bug**
When setting up a  many-to-many relation between to entities: A relation table is created (as expected) and the relation seems to work properly. If strapi reboots it will not start again, giving this error:
```
[2019-06-08T17:04:37.306Z] debug ⛔️ Server wasn't able to start properly.
[2019-06-08T17:04:37.307Z] error error: relation "entity1_entity2__entity2_entity1" already exists
```

**Steps to reproduce the behavior**
1. Create many-to-many relation between two entities (relation table is created as expected)
2. shut down strapi
3. start up strapi again
__error message from above__

If i remove the relation table,Ii am able to start up strapi again. The relations is then (of course) lost between the entities.

**Expected behavior**
strapi should start, even with relations defined.

**System**
Node.js version: v10.15.3
NPM version: 6.4.1
Strapi version: 3.0.0@beta.3
Database: postgres
Operating system: MacOS

**Additional context**
Not sure if related, but the relation table gets a name that exceeds postgres 63 byte name limit. The name is hence truncated upon creation.

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Keeps extra options set in the model definition  file for relation  on CTB update

### Related issue(s)/PR(s)

Fixes #3409

### CODE REVIEW

File: packages/strapi-plugin-content-type-builder/services/schema-builder/content-type-builder.js [hunk 1]
```diff
@@ -158,7 +158,8 @@ module.exports = function createComponentBuilder() {
             this.unsetRelation(oldAttribute);
           }
 
-          newAttribute.autoPopulate = newAttribute.autoPopulate || oldAttribute.autoPopulate;
+          // keep extra options that were set manually on oldAttribute
+          _.defaults(newAttribute, oldAttribute);
```
Comment (petersg83) [inline packages/strapi-plugin-content-type-builder/services/schema-builder/content-type-builder.js:162]: Is it a bug fix too? We couldn't disable the autoPopulate before?

**Review by petersg83 — APPROVED** [review]
LGTM, I have question though

File: packages/strapi-plugin-content-type-builder/services/schema-builder/content-type-builder.js (hunk 1 above)
Comment (alexandrebodin) [reply packages/strapi-plugin-content-type-builder/services/schema-builder/content-type-builder.js:162]: just not required anymore as it is implement with the _.defaults under it :)

Comment (petersg83) [reply packages/strapi-plugin-content-type-builder/services/schema-builder/content-type-builder.js:162]: But this:
```js
newAttribute.autoPopulate = newAttribute.autoPopulate || oldAttribute.autoPopulate;
```
Would always be true if it was true in the old attribute right ?

Comment (alexandrebodin) [reply packages/strapi-plugin-content-type-builder/services/schema-builder/content-type-builder.js:162]: Yes but it cannot be defined in the newAttribute the code was made just to keep the value if it was set in the oldAttribute so the _defaults is good enough to replace it :)

**Comment by ScottAgirs** [comment]
If I may, please - does this basically mean that now we can not worry about the length of Collection Type name length?

**Comment by petersg83** [comment]
Hi @ScottAgirs. Not really, you still have to be careful. However if you encounter a problem with the length, the solution is to customize the name of the table (https://strapi.io/documentation/developer-docs/latest/development/backend-customization.html#relations tab `Many-to-many`)

**Comment by ScottAgirs** [comment]
Hey @petersg83 thought so : ) Thanks for that!

**Pull request merged**